src/facenet_models/__init__.py

```diff
@@ -1,29 +1,43 @@
+from typing import NamedTuple, Optional
+
 import numpy as np
 import torch
 from facenet_pytorch import MTCNN, InceptionResnetV1
 from facenet_pytorch.models.utils.detect_face import crop_resize
 
+__all__ = ["FacenetModel"]
+
+
+class _Detections(NamedTuple):
+    boxes: np.ndarray
+    probabilities: np.ndarray
+    face_landmarks: np.ndarray
+
 
 class FacenetModel:
-    def __init__(self, device=None):
+    def __init__(self, device: Optional[str, torch.device] = None):
```

Comment:
```suggestion
    def __init__(self, device: Optional[Union[str, torch.device]] = None):
```

Reply:
but should this actually just be
```suggestion
    def __init__(self, device: Optional[str] = None):
```
? below we're checking if it's `in {"cuda", "cpu"}`

Reply:
Ah you are right